src/common/charts/shared.js

```diff
@@ -1,99 +1,99 @@
-            series[1].lineColor = series[1].borderColor = chartPrimaryColor;
-            break;
-        case 3:
-            colors = colorsThree;
-            series[0].lineColor = series[0].borderColor = chartSecondaryColor;
-            series[1].lineColor = series[1].borderColor = chartPrimaryColor;
-            series[2].lineColor = series[2].borderColor = chartTertiaryStrokeColor;
-            break;
-        default:
-            colors = colorsMore;
-            series[0].lineColor = series[0].borderColor = chartSecondaryColor;
-            series[1].lineColor = series[1].borderColor = chartTertiaryStrokeColor;
-            series[2].lineColor = series[2].borderColor = chartPrimaryColor;
-            if (series.length > 3) {
-                series[3].lineColor = series[3].borderColor = chartQuaternaryStrokeColor;
-            }
-            if (series.length > 4) {
-                series[4].lineColor = series[4].borderColor = chartQuaternaryStrokeColor;
-            }
-    }
-    return colors;
-}
+    // configure the color sequences based on how many series are used
+    colorsSingle = [chartPrimaryColor],
+    colorsTwo = [chartSecondaryColor, chartPrimaryColor],
+    colorsThree = [chartSecondaryColor, chartPrimaryColor, pattern1],
+    colorsMore = [
+        chartSecondaryColor,
+        pattern1,
+        chartPrimaryColor,
+        pattern2,
+        chartQuinaryBackgroundColor,
+    ],
+    // function is used to set up the colors including lineColor(svg stroke) on each of the series objects
+    // based on the length of the series array
+    setSeriesColors = function (series) {
+        let colors;
+        switch (series.length) {
+            case 1:
+                colors = colorsSingle;
+                series[0].lineColor = series[0].borderColor = chartPrimaryColor;
+                break;
+            case 2:
+                colors = colorsTwo;
+                series[0].lineColor = series[0].borderColor = chartSecondaryColor;
+                series[1].lineColor = series[1].borderColor = chartPrimaryColor;
+                break;
+            case 3:
+                colors = colorsThree;
+                series[0].lineColor = series[0].borderColor = chartSecondaryColor;
+                series[1].lineColor = series[1].borderColor = chartPrimaryColor;
+                series[2].lineColor = series[2].borderColor = chartTertiaryStrokeColor;
+                break;
+            default:
+                colors = colorsMore;
+                series[0].lineColor = series[0].borderColor = chartSecondaryColor;
+                series[1].lineColor = series[1].borderColor = chartTertiaryStrokeColor;
+                series[2].lineColor = series[2].borderColor = chartPrimaryColor;
+                if (series.length > 3) {
+                    series[3].lineColor = series[3].borderColor = chartQuaternaryStrokeColor;
+                }
+                if (series.length > 4) {
```

Comment:
this was done to prevent code duplication of these lines

colors = colorsMore;
                series[0].lineColor = series[0].borderColor = chartSecondaryColor;
                series[1].lineColor = series[1].borderColor = chartTertiaryStrokeColor;
                series[2].lineColor = series[2].borderColor = chartPrimaryColor;
                
                which are the same for both cases 3 and 4